lib/to-interpolated-regexp.js

```diff
@@ -3,7 +3,7 @@ module.exports = function(source) {
     var splitSource = source.split('{componentName}');
     return function(componentName) {
       try {
-        return new RegExp(splitSource.length > 1 ? splitSource[0] + componentName + splitSource[1] : splitSource[0]);
+        return new RegExp(splitSource.join(componentName));
```

Comment:
I'm not sure this logic is right ... Can you please provide tests for this change, then we'll know whether it works correctly?